### PR TITLE
Improve DUMP_READ_OBJECT output

### PR DESCRIPTION
**Pull request by chqrlie**
- improve `JS_DumpString`: use `L` prefix for wide strings
- dump variable kind and flags for locals and closures
- disassemble byte code in DUMP_READ_OBJECT
- pass start_pos to `dump_byte_code` and `dump_single_byte_code`
- write constant pool before function bytecode (bump version to 11)
- update generated code

### CODE REVIEW

File: quickjs.c [hunk 1]
```diff
@@ -27706,25 +27714,21 @@ static void dump_byte_code(JSContext *ctx, int pass,
...
-                printf(" %u:%u", addr, label_slots[addr].pos);
-            if (pass == 2)
-                printf(" %u:%u", addr, label_slots[addr].pos2);
-            if (pass == 3)
-                printf(" %u", addr + pos);
-            break;
         case OP_FMT_label_u16:
             addr = get_u32(tab + pos);
+        has_addr1:
             if (pass == 1)
```
Comment (saghul) [inline quickjs.c:27720]: any reason not tuse if elseif here?

File: quickjs.c [hunk 2]
```diff
@@ -33949,6 +33960,7 @@ static JSValue JS_ReadFunctionTag(BCReaderState *s)
 
     if (local_count != 0) {
         bc_read_trace(s, "vars {\n");
+        bc_read_trace(s, "off flags scope name\n");
```
Comment (saghul) [inline quickjs.c:33963]: should this have `//` ?

**Review by saghul — APPROVED** [review]
LGTM but I'll admit I'm out of my depth here.

File: quickjs.c (hunk 1 above)
Comment (chqrlie) [reply quickjs.c:27720]: <strike>No reason indeed, will modify.</strike>
Actually, it looks ugly with `else if` and worse with a `switch` statement.  The current code is more compact and performance is not an issue.

File: quickjs.c (hunk 2 above)
Comment (chqrlie) [reply quickjs.c:33963]: The output is not really compilable, so no need for comment marks here IMHO.
Try compiling with `DUMP_READ_OBJECT` and look at the output when starting REPL (`build/qjs -d100`), it gives a detailed view of the objects and the bytecode.

Comment (saghul) [reply quickjs.c:33963]: Gotcha, thanks for explaining!

**Pull request merged**